flexget/plugins/input/json.py

```diff
@@ -0,0 +1,137 @@
+import dateutil.parser as parser
+
+import re
+import os
+import logging
+import glob
+
+from flexget import plugin
+from flexget.utils import json
+from flexget.entry import Entry
+from flexget.event import event
+
+log = logging.getLogger('json')
+
+
+class Json(object):
+    """
+    Parse a json file for entries using regular expression.
+
+    :: 
+
+      files: <path to JSON file(s)>
+      encoding: <JSON encoding>
+      entry:
+        - <field>: <corresponding JSON key>
+        - <field>
+        - get_remnants
+
+    Note: each entry must have at least two fields, 'title' and 'url'. If not specified in the config, 
+    this plugin asssumes that keys named 'title' and 'url' exist within the JSON.
+
+    Example::
+
+      json:
+        files: entries.json
+        encoding: utf8
+        entry:
+          - title: name
+          - url: 'web_.*'
+          - get_remnants
+          
+    """
+    
+    schema = {
+        'type': 'object',
+        'properties': {
+            'files': {'type': 'string'},
+            'encoding': {'type': 'string', 'default': 'utf-8'},
+            'entry': {
+                'type': 'array',
+                'items': {
+                    'oneOf': [
+                        {'type': 'string'},
+                        {
+                            'type': 'object',
+                            'additionalProperties': {
+                                'type': 'string', 'format': 'regex'
+                            },
+                        },
+                    ]
+                },
+            },
+        },
+        'required': ['files'],
+        'additionalProperties': False,
+    }
+                    
+    def dt(val):
+        try:
+            return parser.parse(val)
+        except (ValueError, OverflowError):
+            return val
+
+    def on_task_input(self, task, config):
+        files = os.path.expanduser(config['files'])
+        json_encoding = config['encoding']
+        entry_config = config.get('entry')
+        
+        fields = {}
+        get_remnants = False
+        if config.get('entry'):
+            for required_field in config.get('entry'):
+                if isinstance(required_field, str):
+                    if required_field == 'get_remnants':
+                        get_remnants = True
+                    else:
+                        fields[required_field] = re.compile('^' + required_field + '$')
+                else:
+                    fields[next(iter(required_field))] = re.compile(
+                        "^" + required_field[next(iter(required_field))] + "$")
+                        
+            if 'title' not in fields.keys():
+                fields['title'] = re.compile("^title$")
+            if 'url' not in fields.keys():
+                fields['url'] = re.compile("^url$")
+
+        entries = []
+        entry = Entry()
+        
+        list_of_files = glob.glob(files)
```

Comment:
wouldn't this just fetch all files? how are you limiting it to json exactly? also, using pathlib here would be way nicer

Reply:
The way it is setup now, the user can specify either a json file: `/home/username/json_files/my_file.json` or a directory containing all of the json files by using: `/home/user/json_files/*.json`. I added an check to make sure that `json` is in the `files` field value. But I'm open to further suggestions.

Reply:
I think you should force `files` to be in the strings, just try to serialize each file you iterate on, catch error and continue if it doesn't work

Reply:
> I think you should force `files` to be in the strings

Do you mean force the string "json" to be within `files`? Otherwise, I'm not sure that I follow what you are saying.